runtime-testsuite/test/org/antlr/v4/test/runtime/BaseRuntimeTest.java

```diff
@@ -426,35 +420,39 @@ public static ErrorQueue antlrOnString(String workdir,
 	public static UniversalRuntimeTestDescriptor readDescriptor(String dtext)
 			throws RuntimeException
 	{
-		String[] fileSections = {
-				"notes","type","grammar","slaveGrammar","start","input","output","errors",
-				"flags","skip"};
-		Set<String> sections = new HashSet<>(Arrays.asList(fileSections));
 		String currentField = null;
-		String currentValue = null;
+		StringBuilder currentValue = new StringBuilder();
 
 		List<Pair<String, String>> pairs = new ArrayList<>();
-		String[] lines = dtext.split("\r?\n");
-		for (String line : lines) {
-			if ( line.startsWith("[") && line.length()>2 &&
-				 sections.contains(line.substring(1, line.length() - 1)) ) {
-				if ( currentField!=null ) {
-					pairs.add(new Pair<>(currentField,currentValue));
+
+		Matcher matcher = linePattern.matcher(dtext);
```

Comment:
hiya. not sure i find the complex regex more readable than my simple section detector.

Reply:
Reader has to go find and then understand that hairball haha